src/main/java/com/example/SOPSbackend/service/DoctorService.java

```diff
@@ -48,8 +49,11 @@ public void addVaccinationSlot(DoctorEntity doctor, Instant date) {
             throw new InternalValidationException(Map.of("date", "Invalid date value: date should not be set in past"));
 
         VaccinationSlotEntity newSlot = new VaccinationSlotEntity(doctor, transformedDate);
-        //if(vaccinationSlotRepository.findResultsByDate(transformedDate).stream().findAny().orElse(null) != null)
-            vaccinationSlotRepository.save(newSlot);
+        if(vaccinationSlotRepository.findResultsByDateAndDoctor(transformedDate, doctor).stream().findAny().orElse(null) != null)
```

Comment:
Na przyszłość, zamiast `findAny().orElse(null) != null` piszemy `findAny().isPresent()`. A to dlatego, że `findAny()` zwraca `Optional`. `Optional` pojawia się w springboot często. Tutaj: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#isPresent-- można poczytać co taki `Optional` potrafi :) 

IMO

Reply:
Ok, z jakiegoś powodu myślałem że findAny() jest czym innym, mb